src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -118,9 +118,11 @@ public void OnStartActivity(Activity activity, object payload)
         if (textMapPropagator is not TraceContextPropagator)
         {
             var ctx = textMapPropagator.Extract(default, request, HttpRequestHeaderValuesGetter);
-
+            var activityContext = new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true);
             if (ctx.ActivityContext.IsValid()
-                && ctx.ActivityContext != new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true))
```

Comment:
`ActivityContext` equals check also to check if the extracted context's `IsRemote` field is set to `true`.

@cijothomas How do we want to handle that?

Reply:
When context is extracted, it'll always be IsRemote=true....

Reply:
That depends on the implementation of the propagator, right? A user could use a custom propagator which does not set `IsRemote` to `true`.

Reply:
That’ll be called a propagator with a remote bug 😆

Reply:
My point is that this would be a change in how we treat those buggy propagators. Up until now, we would have created a sibling activity with `parentContext.IsRemote == false` for such buggy propagators. With this change we would not do that.

Do you think we should document this change in behavior? If we think that it was bug on our side to consider `IsRemote` for determining whether a sibling activity should be created, we should call it a bug fix in the CHANGELOG.

Reply:
We can probably add it in changelog, but I feel its going to make the changelog too complex to follow. `IsRemote`, by definition, should be `true`, when Propagators (*any*) are used. Maybe we need to explore a way to prevent IsRemote being set to false in Propagators somehow. (It should not be a priority, just mentioning.)